numba_dpex/core/typing/typeof.py

```diff
@@ -65,3 +66,9 @@ def typeof_usm_ndarray(val, c):
         queue=val.sycl_queue,
         addrspace=address_space.GLOBAL,
     )
+
+
+@typeof_impl.register(DpnpNdarray)
+def typeof_dpnp_ndarray(val, c):
+
+    return typeof_usm_ndarray(val, c)
```

Comment:
After think a bit more about it, we need to do one more thing. `typeof_usm_ndarray` will return a `USMNdarray` object. But, we have to convert it to a `DpnpNdarray` object. Otherwise, `typeof` will always return `USMNdarray` and not the `DpnpNdarray` type.

Reply:
Any example of converting from USMNdarray to DpnpNdarray?

Reply:
Since, Python has no notion of casting. I suggest creating a helper function that implements the logic inside `typeof_usm_ndarray` and that function is used by both typeof functions.